File: frontend/src/types/project.ts
export type ProjectStatus = 'planning' | 'active' | 'completed' | 'on_hold';

export interface Project {
  id: string;
  code: string;
  name: string;
  description?: string;
  status: ProjectStatus;
  start_date?: string;
  end_date?: string;
  budget?: number;
  created_at: string;
  updated_at: string;
}

export interface ProjectListItem extends Project {
  activities_count: number;
  total_spend: number;
  evidence_completeness: number;
}

export interface ProjectDetail extends Project {
  scientific_rationale?: string;
  eligibility_notes?: string;
  activities_count: number;
  total_spend: number;
  evidence_status: EvidenceStatus;
}

export interface ProjectCreate {
  code: string;
  name: string;
  description?: string;
  status?: ProjectStatus;
  start_date?: string;
  end_date?: string;
  budget?: number;
  scientific_rationale?: string;
  eligibility_notes?: string;
  eligibility_rationale?: string;
  notes?: string;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  status?: ProjectStatus;
  start_date?: string;
  end_date?: string;
  budget?: number;
  scientific_rationale?: string;
  eligibility_notes?: string;
  eligibility_rationale?: string;
  notes?: string;
}

export interface RdActivity {
  id: string;
  project_id: string;
  title: string;
  description?: string;
  activity_date: string;
  hours?: number;
  personnel?: string;
  methodology?: string;
  results?: string;
  notes?: string;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface RdActivityCreate {
  title: string;
  description?: string;
  activity_date: string;
  hours?: number;
  personnel?: string;
  methodology?: string;
  results?: string;
  notes?: string;
}

export interface RdActivityUpdate {
  title?: string;
  description?: string;
  activity_date?: string;
  hours?: number;
  personnel?: string;
  methodology?: string;
  results?: string;
  notes?: string;
}

export interface SpendSummary {
  project_id: string;
  categories: {
    salaries: number;
    cro_contractor: number;
    consumables: number;
    equipment: number;
    other: number;
  };
  total: number;
}

export interface EvidenceStatus {
  project_id: string;
  has_activities: boolean;
  activities_count: number;
  has_transactions: boolean;
  transactions_count: number;
  has_evidence_files: boolean;
  evidence_files_count: number;
  completeness_percentage: number;
}

export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/schemas/project.py
from datetime import datetime, date
from decimal import Decimal
from typing import Optional, List
from uuid import UUID
from pydantic import BaseModel, Field

from app.models.project import ProjectStatus


class ProjectBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None
    status: ProjectStatus = ProjectStatus.PLANNING
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    budget: Optional[Decimal] = None
    eligibility_rationale: Optional[str] = None
    notes: Optional[str] = None


class ProjectCreate(ProjectBase):
    code: str = Field(..., min_length=1, max_length=50)
    scientific_rationale: Optional[str] = None
    eligibility_notes: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    status: Optional[ProjectStatus] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    budget: Optional[Decimal] = None
    eligibility_rationale: Optional[str] = None
    notes: Optional[str] = None
    scientific_rationale: Optional[str] = None
    eligibility_notes: Optional[str] = None


class ProjectResponse(BaseModel):
    id: UUID
    code: str
    name: str
    description: Optional[str] = None
    status: ProjectStatus
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    budget: Optional[Decimal] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ProjectStats(BaseModel):
    activities_count: int
    total_spend: Decimal
    evidence_completeness: float  # Percentage 0-100


class ProjectDetail(ProjectResponse):
    scientific_rationale: Optional[str] = None
    eligibility_notes: Optional[str] = None
    activities_count: int
    total_spend: Decimal
    evidence_status: dict


class ProjectListItem(ProjectResponse):
    activities_count: int
    total_spend: Decimal
    evidence_completeness: float
